src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -307,6 +307,20 @@ private static bool IsFloatPointExpressionOfUnknownPrecision(BoundExpression rew
 
                     break;
 
+                case ConversionKind.MethodGroup:
+
+                    if (_inExpressionLambda || explicitCastInCode || (object)symbolOpt == null || !symbolOpt.IsStatic || symbolOpt.MethodKind != MethodKind.Ordinary)
+                    {
+                        break;
+                    }
+                    if (symbolOpt.IsExtensionMethod && ((BoundMethodGroup)rewrittenOperand).ReceiverOpt.Kind != BoundKind.TypeExpression)
```

Comment:
@AlekseyTs Good catch, didn't think twice when the `obj.M`, `Extensions.M` case failed me, the method could be in the same extensions class such that `M` would be used.

@agocke Sure, that could be useful.